models/templates/llama-cpp-microsoft-Phi-4-mini-instruct.jinja

```diff
@@ -0,0 +1,37 @@
+{%- if messages[0]["role"] == "system" %}
+    {%- set system_message = messages[0]["content"] %}
+{% elif tools is defined -%}
+    {%- set system_message = "" -%}
+{%- endif %}
+
+{%- if system_message is defined -%}
+    {{- '<|system|>' + system_message -}}
+    {%- if tools is defined -%}
+        {% for tool in tools %}
+            {{- '<|tool|>' + (tool['function'] | tojson) + '<|/tool|>' -}}
+        {% endfor %}
+        {%- if '<|tool_call|>' not in system_message -%}
+            {{- 'You are a helpful assistant with some tools.\nTo use a tool, respond in this format: <|tool_call|>{"name": "foo", "arguments": {"a": 1}}<|/tool_call|>' -}}
+        {%- endif -%}
+    {%- endif -%}
+    {{- '<|end|>' -}}
+{%- endif -%}
+{%- for message in messages -%}
+    {%- if message['role'] == 'tool' -%}
+        {{- '<|tool_response|>' + (message['content'] | tojson) + '<|end|>' -}}
```

Comment:
This is seems to be the cause of the test_calc_result failure. Good news is, the following (wild hunch / inspired by other tool call styles) makes it work:

```
{{- '<|tool_response|>' + message['name'] + '<|tag|>' + (message['content'] | tojson) + '<|end|>' -}}
```